wallet/wallet.go

```diff
@@ -1305,13 +1311,13 @@ func (w *Wallet) TxRecord(txSha *wire.ShaHash) (r *txstore.TxRecord, ok bool) {
 	return nil, false
 }
 
-// openWallet opens a wallet from disk.
-func openWallet() (*Wallet, error) {
-	netdir := networkDir(cfg.DataDir, activeNet.Params)
-	dbPath := filepath.Join(netdir, walletDbName)
+// OpenWallet opens a wallet from disk.
+func OpenWallet(config *Config) (*Wallet, error) {
+	netdir := NetworkDir(config.DataDir, config.ChainParams)
```

Comment:
As mentioned elsewhere, I'd prefer all of this be up to the caller and this function takes an already opened `walletdb.DB` to work with.